src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets

```diff
@@ -63,8 +63,7 @@ The .NET Foundation licenses this file to you under the MIT license.
 
     <ItemGroup>
       <NetCoreAppNativeLibrary Include="System.Native" />
-      <!-- FIXME: The library is currently not available for iOS-like platforms -->
-      <NetCoreAppNativeLibrary Include="System.Globalization.Native" Condition="'$(StaticICULinking)' != 'true' and '$(_IsiOSLikePlatform)' != 'true'" />
+      <NetCoreAppNativeLibrary Include="System.Globalization.Native" Condition="'$(StaticICULinking)' != 'true'" />
```

Comment:
@filipnavara, in case it wasn't obvious: the `StaticICULinking` is a bit different. It is meant to run cmake _during_ `dotnet publish -p:PublishAot=true` on user machine. I'm not sure how the variant added in `src/installer/pkg/sfx/Microsoft.NETCore.App/Directory.Build.props` will be used here, but we seem to have two other options:

1. either bring iOS to the same plan as existing platforms (run cmake to build Globalization shim during dotnet-publish on user machine).
2. or use it from the framework directory; change line 89 as:
    ```diff
    -       <NativeLibrary Include="$(IntermediateOutputPath)libs/System.Globalization.Native/build/libSystem.Globalization.Native.a" />
    +       <NativeLibrary Include="$(IntermediateOutputPath)libs/System.Globalization.Native/build/libSystem.Globalization.Native.a" Condition="'$(_IsiOSLikePlatform)' != 'true'" />
    +       <NativeLibrary Include="$(IlcFrameworkNativePath)libSystem.Globalization.Native.a" Condition="'$(_IsiOSLikePlatform)' == 'true'" />
    ```
      * Note that we already have (an unused!) copy of `libSystem.Globalization.Native.a` in the framework directory:
          ```zsh
          % ls ~/.nuget/packages/runtime.linux-arm64.microsoft.dotnet.ilcompiler/8.0.0-preview.2.23116.1/framework/*Globalization*.a
          /Users/am11/.nuget/packages/runtime.linux-arm64.microsoft.dotnet.ilcompiler/8.0.0-preview.2.23116.1/framework/libSystem.Globalization.Native.a
          ```
          which comes from https://github.com/dotnet/runtime/blob/af18fd07d1bb5a0d923a49c39b19c6fbcb6dc07b/eng/liveBuilds.targets#L175 (unused files are  missing from the ilc package excludes, which should be cleaned up separately)

I'm not familiar with NativeAOT step in Xamrian publish workflow, but if option 1 is feasible for iOS-like platforms, I'd go for it.

Reply:
libSystem.Globalization.Native.a was not built for iOS-like platforms, so that's part of the fix.

The final linking is done either through AppleAppBuiler or the Xamarin tooling. Both of them can consume the .a file once it exists.

`StaticICULinking` remains unused for the moment for these platforms.

Reply:
> `StaticICULinking` remains unused for the moment for these platforms.

I missed that! 😁

> libSystem.Globalization.Native.a was not built for iOS-like platforms, so that's part of the fix.

Cool. For NativeAOT, does Xamarin use shared libraries from `microsoft.dotnet.ilcompiler` package or from the Microsoft.NETCore.App / shared framework? e.g. `libSystem.Globalization.Native.a` _can_ be different if user has csproj file with:

```csproj
<!-- Different version than what comes with SDK https://github.com/dotnet/installer/blob/8ce7bb903a8d818bd288267f82be0d98f5e6c757/src/redist/targets/GenerateBundledVersions.targets#L408 -->
<PackageReference Include="Microsoft.DotNet.ILCompiler" Version="$(DifferentVersion)" />
```

Reply:
That's kind of unanswered packaging question at the moment. 😅

For the current MonoVM-based workloads it comes from Microsoft.NETCoreApp.Runtime.Mono.<RID> packages. This PR adds libSystem.Globalization.Native.a into the package and removes the same APIs from libmonosgen.a (or whether is the static runtime name).

Reply:
> StaticICULinking remains unused for the moment for these platforms.
Should we add a MSBuild warning/error when someone tries to use it on these platforms?

Reply:
> Should we add a MSBuild warning/error when someone tries to use it on these platforms?

It would be nice to have explicit check but I would prefer not to mix it into this PR.

I don't touch any of the `StaticICULinking` code here and it's documented to work only on Linux when not cross-compiling:
https://github.com/dotnet/runtime/blob/cabaab5e90e19256ae404970091350bb6f0f7139/src/coreclr/nativeaot/docs/compiling.md?plain=1#L77-L85